### ISSUE TITLE
UTF-8 encoding issue

### ISSUE THREAD

**Issue by AyubOpen** [brettwooldridge/HikariCP#932]
#### Environment
```
HikariCP version:2.6.2
JDK version     : 1.8.0_102
Database        : Oracle
Driver version  : 11.2.0.3
```
-----------------------------------------------------------------------------------------

@brettwooldridge  I am trying the same appoarch as you mentioned about config.add for utf-8, but I still see the issue of garbled characters reading from oracle database. Code works if I use plain jdbc; below is my property file. complete code is on github https://github.com/AyubOpen/spring-boot-jdbc

```
//hikari.properties
driverClassName=oracle.jdbc.driver.OracleDriver
jdbcUrl=jdbc:oracle:thin:@localhost:1521/xe
maximumPoolSize=20
username=testdbapp
password=oracle
dataSource.cachePrepStmts=true
dataSource.prepStmtCacheSize=250
```
```java
@configuration
public class AppConfig {

   @Bean(destroyMethod = "close")
   public DataSource dataSource() throws SQLException {
      HikariConfig config = new HikariConfig("/hikari.properties");
      config.addDataSourceProperty("characterEncoding","utf8"); //tried with UTF-8 as well
      config.addDataSourceProperty("useUnicode","true");
      HikariDataSource dataSource = new HikariDataSource(config);

      return dataSource;
   }
}
```
this is a spring boot application, please advise

**Comment by brettwooldridge** [comment]
A couple of things...

``dataSource.cachePrepStmts`` and ``dataSource.prepStmtCacheSize`` are properties from the MySQL example on the project page, they will have no effect on Oracle.  The same is true of ``useUnicode`` and ``characterEncoding`` -- those are both MySQL properties, not Oracle properties.

Finally, I ran your code and it worked fine.  I deleted the above elements from ``hikari.properties`` and ``AppConfig.java`` -- but it shouldn't matter.  I created a table called ``pending_menu_item`` and inserted two rows with Japanese characters:

```sql
INSERT INTO pending_menu_item (menu_item_id, menu_item_name) VALUES (1, '魚');
INSERT INTO pending_menu_item (menu_item_id, menu_item_name) VALUES (2, 'おいしいラメン');
```

I modified ``CustomerRepository.java`` to remove the WHERE clause.

Here is the run:
```
 ~/Documents/dev/spring-boot-jdbc$ mvn spring-boot:run -Drun.arguments="display"
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building spring-boot-jdbc 1.0
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] >>> spring-boot-maven-plugin:1.5.1.RELEASE:run (default-cli) > test-compile @ spring-boot-jdbc >>>
[INFO]
[INFO] --- maven-resources-plugin:3.0.2:resources (default-resources) @ spring-boot-jdbc ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] Copying 2 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ spring-boot-jdbc ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-resources-plugin:3.0.2:testResources (default-testResources) @ spring-boot-jdbc ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/brettw/Documents/dev/spring-boot-jdbc/src/test/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ spring-boot-jdbc ---
[INFO] No sources to compile
[INFO]
[INFO] <<< spring-boot-maven-plugin:1.5.1.RELEASE:run (default-cli) < test-compile @ spring-boot-jdbc <<<
[INFO]
[INFO] --- spring-boot-maven-plugin:1.5.1.RELEASE:run (default-cli) @ spring-boot-jdbc ---
getMaximumPoolSize = {}
Display items...
魚
おいしいラメン
Done!
```

I *did* run Oracle XE [via docker](https://hub.docker.com/r/sath89/oracle-12c/), and I set these docker environment variables:

```
docker run -d -p 8080:8080 -p 1521:1521 -v data:/u01/app/oracle \
  -e USE_UTF8_IF_CHARSET_EMPTY=true -e CHARACTER_SET=AL32UTF8 sath89/oracle-12c
```

*I do not know how that translates into general Oracle settings*, I suspect those environment variables are specific to that Docker image.

Basically, from what I understand reading the internet, if the database is creating in Oracle with the proper encoding, there is nothing to configure on the Java/JDBC side.

I don't know what else to say.

**Comment by AyubOpen** [comment]
Thank you for checking, the only difference is that I am connecting to Oracle running on Virtualbox. Plain JDBC works, when using datasource I face the issue with encoding.

Thank you for the clues, I will try to debug further and post my finding here

**Comment by AyubOpen** [comment]
I deployed the same war file on ubuntu running tomcat and it seems to show arabic text correctly. Might be some combination on windows + tomcat + springboot
